apps/kol-amelamdim/pages/api/upload-file.ts

```diff
@@ -40,16 +49,32 @@ export default async function handler(req, res) {
         Bucket: process.env.AWS_BUCKET_NAME,
         Key: fileLocation,
         Body: contents,
-        Tagging: 'reviewed=false',
+        ACL: 'public-read',
       };
 
-      s3.upload(params, (err, _) => {
-        if (err) {
-          res.status(400).json(API_ERRORS.uploadFileError);
-        }
-
-        res.status(200).json({ isUploaded: true });
-      });
+      const returnedData = s3
+        .upload(params, (err, _) => {
+          if (err) {
+            res.status(400).json(API_ERRORS.uploadFileError);
+          }
+        })
+        .promise();
+      return returnedData
+        .then(async (response) => {
+          await connect();
+          await File.create({
+            key: uuidv4(),
```

Comment:
Why we need it? mongoose know automatically to add unique _id for every document

Reply:
if we does not use some special key its fails for uploading same file with the same name..